### PR TITLE
[fix](nereids)should normalize window expression by bottom project's output in ExtractAndNormalizeWindowExpression rule

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: regression-test/data/nereids_tpcds_shape_sf1000_p0/shape/query70.out [hunk 1]
```diff
@@ -32,23 +32,22 @@ PhysicalResultSink
...
-----------------------------------------------------------PhysicalProject
-------------------------------------------------------------hashJoin[INNER_JOIN] hashCondition=((store.s_store_sk = store_sales.ss_store_sk)) otherCondition=() build RFs:RF1 s_store_sk->[ss_store_sk]
---------------------------------------------------------------hashJoin[INNER_JOIN] hashCondition=((date_dim.d_date_sk = store_sales.ss_sold_date_sk)) otherCondition=() build RFs:RF0 d_date_sk->[ss_sold_date_sk]
-----------------------------------------------------------------PhysicalProject
-------------------------------------------------------------------PhysicalOlapScan[store_sales] apply RFs: RF0 RF1
-----------------------------------------------------------------PhysicalDistribute[DistributionSpecReplicated]
-------------------------------------------------------------------PhysicalProject
---------------------------------------------------------------------filter((date_dim.d_month_seq <= 1231) and (date_dim.d_month_seq >= 1220))
-----------------------------------------------------------------------PhysicalOlapScan[date_dim]
+--------------------------------------------PhysicalQuickSort[LOCAL_SORT]
```
Comment (924060929) [inline regression-test/data/nereids_tpcds_shape_sf1000_p0/shape/query70.out:35]: Why the origin shape missing PhysicalQuickSort?

Comment (starocean999) [reply regression-test/data/nereids_tpcds_shape_sf1000_p0/shape/query70.out:35]: origin shape use two phase PhysicalPartitionTopN with a data shuffle between them. This pr correct the slot id issue, so the new shape use one phase PhysicalPartitionTopN + PhysicalQuickSort instead, and no need to shuffle data.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ExtractAndNormalizeWindowExpression.java [hunk 2]
```diff
@@ -91,7 +92,17 @@ public Rule build() {
...
+
+            // because alias is pushed down to bottom project
+            // we need replace alias's child expr with corresponding alias's slot in output
+            // so create a customNormalizeMap alias's child -> alias.toSlot to do it
+            Map<Expression, Slot> customNormalizeMap = toBePushedDown.stream()
+                    .filter(expr -> expr instanceof Alias)
+                    .collect(Collectors.toMap(expr -> ((Alias) expr).child(), expr -> ((Alias) expr).toSlot(),
+                            (oldExpr, newExpr) -> oldExpr));
+
+            List<NamedExpression> normalizedOutputs1 = context.normalizeToUseSlotRef(outputs,
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ExtractAndNormalizeWindowExpression.java:104]: could we change normalizedOutputs1 to a better name?

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**